### ISSUE TITLE
Does this package calling https://github.com/acheong08/ChatGPT-Proxy-V4 under the hood?

### ISSUE THREAD

**Issue by hongyi-zhao** [acheong08/ChatGPT-to-API#80]
IMO, the following two packages are wonderful and key tools when working with OpenAI API:

https://github.com/acheong08/ChatGPT-to-API
https://github.com/acheong08/ChatGPT-Proxy-V4

So, I want to know if the ChatGPT-TO-API calls ChatGPT-PROXY-V4 in the background to further simplify the access workflow?

**Comment by acheong08** [comment]
It doesn't.

You can tell it to by setting `API_REVERSE_PROXY="https://bypass.churchless.tech/conversation"` or something like that

**Comment by hongyi-zhao** [comment]
Then, what is the difference between the following endpoints:
 
https://bypass.churchless.tech/conversation
https://free.churchless.tech/v1/chat/completions

In addition, are there still other endpoints exposed in the domain churchless.tech?

See https://github.com/binary-husky/gpt_academic/issues/900#issuecomment-1615961556 for the related discussion.

**Comment by acheong08** [comment]
> Do you mean that all the endpoints exposed under the domain churchless.tech are constructed based on https://github.com/acheong08/ChatGPT-Proxy-V4?

No. The bypass endpoint can be used if your IP is banned by Cloudflare/OpenAI

> What is the difference between the following endpoints?

Bypass: Gets around cloudflare. API format is identical to chat.openai.com

Free: Gets around cloudflare **and** makes the API format identical to api.openai.com

**Comment by hongyi-zhao** [comment]
> Bypass: Gets around cloudflare. API format is identical to chat.openai.com

If I understand correctly, the endpoint of this method is only applicable to applications that are compatible with web-based access, aka, without using API key.

For example, the https://github.com/binary-husky/gpt_academic works based on the API key, so the `Bypass` endpoint cannot be used directly with it, while the `Free` endpoint does as you have commented [here](https://github.com/binary-husky/gpt_academic/issues/900#issuecomment-1615961556).

**Comment by acheong08** [comment]
> For example, the https://github.com/binary-husky/gpt_academic works based on the API key, so the Bypass endpoint cannot be used directly with it, while the Free endpoint does, as you have commented https://github.com/binary-husky/gpt_academic/issues/900#issuecomment-1615961556.

It's not a matter of whether it uses API key or not. The difference is in the API response format:

```json
{"id":"chatcmpl-QXlha2FBbmROaXhpZUFyZUF3ZXNvbWUK","object":"chat.completion","created":0,"model":"gpt-3.5-turbo-0301","usage":{"prompt_tokens":0,"completion_tokens":0,"total_tokens":0},"choices":[{"index":0,"message":{"role":"assistant","content":"1"},"finish_reason":null}]}
```

vs

```
{"content_type": "text", "parts": ["Hello! How can I assist you today?"]}, "status": 
"finished_successfully", "end_turn": true, "weight": 1.0, "metadata": {"message_type": "next", 
"model_slug": "gpt-4", "finish_details": {"type": "stop", "stop": "<|diff_marker|>"}}, "recipient": 
"all"}, "conversation_id": "dbb3f52a-70a1-4f67-a7ca-3471d524368b", "error": null}
```

**Comment by hongyi-zhao** [comment]
Why the returned content, aka, `["Hello! How can I assist you today?"]`, doesn't appear in the former response?

**Comment by acheong08** [comment]
Oh. It was a different prompt

**Comment by acheong08** [comment]
The response was "1"

**Comment by hongyi-zhao** [comment]
I see, but how can I use https://bypass.churchless.tech/conversation as the endpoint of https://github.com/binary-husky/gpt_academic, which you have commented [here](https://github.com/binary-husky/gpt_academic/issues/900#issuecomment-1615961556)?

**Comment by acheong08** [comment]
cant
